src/libaktualizr/package_manager/packagemanagerfake.cc

```diff
@@ -110,5 +110,10 @@ bool PackageManagerFake::fetchTarget(const Uptane::Target& target, Uptane::Fetch
     return false;
   }
 
+  if (target.IsOstree()) {
```

Comment:
Don't we do a similar check here https://github.com/advancedtelematic/aktualizr/blob/1d9509428441396674ac5b78ec2513de63f99edc/src/libaktualizr/primary/sotauptaneclient.cc#L651 ?

Reply:
Yes, but these two checks do different things. The first (the one you've pointed out) makes sure that we don't uselessly download OSTree commits for Secondaries since we can't do anything with them. The second (the one I've added) is specific to the fake/binary package manager and makes sure that we don't attempt to fetch OSTree commits (for the Primary) when we aren't using the OSTree package manager.

Reply:
> makes sure that we don't attempt to fetch OSTree commits (for the Primary) when we aren't using the OSTree package manager.

I suppose the backend shouldn't allow such an update in the first place, i.e. it should not allow installation of targets type of which is not supported by the backend. 
Also, it could be more generic verification in SotaUptaneClient, something like if (!package_manager_->isTargetSupported()) {} or in the future when the logical ECU is in place then a container/array could expose an API allowing to search for an instance that supports a given target.

Reply:
> I suppose the backend shouldn't allow such an update in the first place, i.e. it should not allow installation of targets type of which is not supported by a device.

Yes, but I don't think the backend processes that information. But even if it did, we still have to check that on the device.

> Also, it could be more generic verification in SotaUptaneClient

Very true, but I don't think we're there yet. This is merely an incremental improvement.